conda_build/metadata.py

```diff
@@ -622,6 +622,7 @@ def parse(data, config, path=None):
         "prelink_message": None,
         "readme": None,
     },
+    "extra": {},
```

Comment:
`extra` could be anything else (scalar or sequence, not just mapping) acc. to https://docs.conda.io/projects/conda-build/en/stable/resources/define-metadata.html#extra-section .
This broke an assumption in `conda-smithy` at least: https://github.com/conda-forge/conda-smithy/pull/1816 .

So, technically a breaking change; but I'd be okay with requiring a mapping here.
This needs an update to the docs, though.

Reply:
Opened gh-5125 to track this.